File: 2163-Kth Distinct String in an Array.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string, int> c;
        int cur = 0;
        for(auto s : arr)
            c[s] += 1;
        
        for (auto v : arr){
            if (c[v]==1){
                cur += 1;
                if (cur==k)
                    return v;
            }
        }
        return "";
    }
};

File: 2163-Kth Distinct String in an Array.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        c = Counter(arr)
        cur = 0
        for i, v in c.items():
            if v == 1:
                cur += 1
                if cur == k:
                    return i
        
        return ''